onnxruntime/contrib_ops/rocm/math/gemm_float8_ck.cuh

```diff
@@ -0,0 +1,276 @@
+    constexpr const uint32_t sign_mask = 0x80008000;
+    constexpr const uint32_t exp_compensate = []() {
+      if constexpr (std::is_same_v<F8, Float8E4M3FN>) {
+        return 0x20002000;
+      } else if constexpr (std::is_same_v<F8, Float8E4M3FNUZ>) {
+        return 0x1c001c00;
+      }
+    }();
+
+    uint32_t xs_u32 = reinterpret_cast<const uint32_t&>(xs);
+    uint32_t x_u32_0 = __byte_perm(xs_u32, 0, 0x1504);
+    uint32_t x_u32_1 = __byte_perm(xs_u32, 0, 0x3726);
+    uint32_t exp_0 = (x_u32_0 & mask) >> 1;
+    uint32_t exp_1 = (x_u32_1 & mask) >> 1;
+    uint32_t v_0 = (x_u32_0 & sign_mask) | (exp_0 + exp_compensate);
+    uint32_t v_1 = (x_u32_1 & sign_mask) | (exp_1 + exp_compensate);
+    uint64_t v = v_0 | uint64_t(v_1) << 32;
+    ys = scale * reinterpret_cast<ck::half4_t&>(v);
+  }
+
+  float scale_value_;
+  const float* const dev_scale_ptr_;
+};
+#endif
+
+namespace blas {
+
+template <typename TA, typename TB, typename TC>
+struct GemmFloat8Params : tunable::OpParams {
+  std::string Signature() const override {
+    return MakeString(BlasOpToString(opa), BlasOpToString(opb), "_", m, "_", n, "_", k);
+  }
+
+  rocblas_handle handle;
+  BlasOp opa;
+  BlasOp opb;
+  int64_t m;
+  int64_t n;
+  int64_t k;
+  float scale_a{};
+  const float* scale_a_dev{};
+  const TA* a;
+  int64_t lda;
+  float scale_b{};
+  const float* scale_b_dev{};
+  const TB* b;
+  int64_t ldb;
+  TC* c;
+  float scale_c{};
+  const float* scale_c_dev{};
+  int64_t ldc;
+};
+
+#if defined(USE_COMPOSABLE_KERNEL) && !defined(DISABLE_FLOAT8_TYPES)
+
+using Row = ck::tensor_layout::gemm::RowMajor;
+using Col = ck::tensor_layout::gemm::ColumnMajor;
+
+using Nop = ck::tensor_operation::element_wise::PassThrough;
+
+void add_device_gemm_xdl_splitk_f8_f16_f16_mk_kn_mn_instances(
+    std::vector<std::unique_ptr<ck::tensor_operation::device::DeviceGemmSplitK<
+        Row, Row, Row, ck::f8_t, ck::half_t, ck::half_t, Scale<Float8E4M3FN>, Nop, Nop>>>& instances);
+
+void add_device_gemm_xdl_splitk_f8_f16_f16_mk_kn_mn_instances(
+    std::vector<std::unique_ptr<ck::tensor_operation::device::DeviceGemmSplitK<
+        Row, Row, Row, ck::f8_t, ck::half_t, ck::half_t, Scale<Float8E4M3FNUZ>, Nop, Nop>>>& instances);
+
+void add_device_gemm_xdl_splitk_f16_f8_f16_mk_kn_mn_instances(
+    std::vector<std::unique_ptr<ck::tensor_operation::device::DeviceGemmSplitK<
+        Row, Row, Row, ck::half_t, ck::f8_t, ck::half_t, Nop, Scale<Float8E4M3FN>, Nop>>>& instances);
+
+void add_device_gemm_xdl_splitk_f16_f8_f16_mk_kn_mn_instances(
+    std::vector<std::unique_ptr<ck::tensor_operation::device::DeviceGemmSplitK<
+        Row, Row, Row, ck::half_t, ck::f8_t, ck::half_t, Nop, Scale<Float8E4M3FNUZ>, Nop>>>& instances);
+
+void add_device_gemm_xdl_splitk_f16_f8_f16_mk_nk_mn_instances(
+    std::vector<std::unique_ptr<ck::tensor_operation::device::DeviceGemmSplitK<
+        Row, Col, Row, F16, F8, F16, Nop, Scale<Float8E4M3FN>, Nop>>>& instances);
```

Comment:
Could you please unify the use of datatype? The above four functions use ck::f8_t/ck::half_t and the bottom two functions use F8/F16.